src/features/Chat/components/ActiveWindow/Header/Buttons.tsx

```diff
@@ -20,38 +23,60 @@ type Props = {
 const Buttons = ({ chat, showSearch }: Props) => {
   const { t } = useTranslation('chat');
 
+  const [dialogVariant, setDialogVariant] = useState<DialogVariant>('archive');
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const closeDialog = () => setIsDialogOpen(false);
+
+  const openDialog = (variant: DialogVariant) => {
+    setDialogVariant(variant);
+    setIsDialogOpen(true);
+  };
+
   const [updateChatStatus] = useUpdateStatusMutation();
   const userId = useAppSelector(selectUserId);
 
   const updateStatus = (status: ChatFolder) => {
+    if (isDialogOpen) closeDialog();
```

Comment:
Is this if necessary?

Reply:
Absolutely! Without this the dialog stays open when you press **archive** etc.

Reply:
Of course the updateStatus function now does more than one thing and the name is misleading so actually I could refactor that to make more sense. 🤔

Reply:
Yeah, but would it not close the dialog with the closeDialog-function call?
We dont need to check if isDialogOpen

Reply:
> 

Yeah, I would call it 
handleUpdateConfirmation

Reply:
handleUpdateStatusConfirmation